src/Microsoft.VisualStudio.Threading.Tests/TplExtensionsTests.cs

```diff
@@ -151,12 +151,47 @@ public void ApplyResultToPreCompletedNonGeneric()
         [Fact]
         public void WaitWithoutInlining()
```

Comment:
1. I don't think we need a test for this. A task can't complete before tasks that have attached to it complete, and that's built in TPL behavior rather than functionality that we're providing.
2. Yes, I'll add these.